.pre-commit-config.yaml

```diff
@@ -0,0 +1,14 @@
+repos:
+  - repo: local
+    hooks:
+      - id: black
+        name: black
+        entry: python -m black
+        language: system
+        types_or: [python, pyi]
+      - id: flakehell
+        name: flakehell
+        entry: python -m flakehell lint
+        language: system
+        types_or: [python, pyi]
+        
```

Comment:
On activerait aussi `trailing-whitespace` (retire les blancs en fin de ligne), `mixed-line-ending` (s'assure que c'est toujours \n à la fin et non pas \r\n comme windows fait), `end-of-file-fixer` (s'assure que le fichier se fini avec une ligne vide à la fin, plus jolie dans `cat`) et `debug-statements` (vérifie qu'on a pas oublié de retirer pdb/breakpoint() des sources avant de commiter) ?

Reply:
Perso je suis OK pour les 4 :)

Reply:
Sur ceux que tu cites:
- trailing-whitespace: black s'en occupe
- mixed-line-ending: black s'en occupe
- end-of-file-fixer: black s'en occupe
- debug-statements: a ajouter

Il y en a d'autres propose directement par pre-commit, voir ici https://pypi.org/project/pre-commit-hooks/, je pense que les suivants sont bien:
- check-merge-conflict: Check for files that contain merge conflict strings.
- check-case-conflict: Check for files with names that would conflict on a case-insensitive filesystem like MacOS HFS+ or Windows FAT.
- detect-private-key: Checks for the existence of private keys.
- destroyed-symlinks: Detects symlinks which are changed to regular files with a content of a path which that symlink was pointing to.
- check-symlinks: Checks for symlinks which do not point to anything.
- check-toml
- check-yaml
- check-json
- check-xml

Vous en pensez quoi ?

Reply:
Vu qu'on va aussi avoir pas mal de template de config (des fichiers pas python), je continue de penser que trailing-whitespace et end-of-file-fixer sont utiles. Concernant les check yaml/xml/json je ne pense pas qu'on en aura besoin, existe-t-il des fichiers de configurations d'outils qu'on compte supporter qui utilisent ces formats ? est-ce qu'on ne pourrait pas plutôt introduire ces linter au moment où ces fichiers sont ajoutés ? Concernant toml (si c'est juste vérifier que le fichier parse correctement et non pas en plus *linter* le fichier) je dirais que nous devrions avoir des tests unitaires qui vérifient qu'incipyt est capable de bootstraper des projets, si le fichier toml est foireux, il devrait être remonté par les tests unitaires.

+1 pour check-merge-conflict, check-case-conflict, detect-private-key (est-ce que celui-ci est vraiment nécessaire??) et destroyed-symlinks

Reply:
Je pensais pas au fichier non-python du coup, en effet ça a un intérêt de remettre les 3 dont black s'occupe.

Pour les checker de config, c'est vrai qu'on pourrait juste les mettre si on en a besoin.

Pour detect-private-key, je dirais qu'en théorie on en a pas besoin, mais je le vois comme un garde-fou. Dans le sens ou contrairement au checker de configs on peut pas vraiment le mettre au besoin : si on sait qu'on a mis une clé privée, ça implique qu'on sait qu'on doit l'enlever.

Reply:
après réflexion je trouve que c'est mieux d'inclure les hooks quand on aura effectivement des fichiers de conf qui utilisent ces formats (pour yaml du coup on peut d'ores et déjà se poser la question, pour la conf des hooks internes à incipyt et de la CI github par exemple)

pas d'avis sur detect-private-key

Reply:
commit 55e5c98